src/ray/common/asio/io_service_pool.cc

```diff
@@ -25,8 +25,8 @@ IOServicePool::~IOServicePool() {}
 void IOServicePool::Run() {
   for (size_t i = 0; i < io_service_num_; ++i) {
     io_services_.emplace_back(
-        std::unique_ptr<boost::asio::io_service>(new boost::asio::io_service));
-    boost::asio::io_service &io_service = *io_services_[i];
+        std::unique_ptr<instrumented_io_context>(new instrumented_io_context));
```

Comment:
Just some thoughts here. Is it better to do things like:

`using ray_io_context = instrumented_io_context;`

and use `ray_io_context` everywhere? So later if we want to do this kind of change again, we don't need to update `instrumented_io_context` everywhere.

Reply:
Also, I tend to grep `class Foo` when I'm trying to look up the definition of some class `Foo` that I'm seeing used somewhere, and I'm sure that `using` definition would throw me for a loop! 😁

Reply:
I agreed that `instrumented_io_context` does show more semantics about the purpose. The benefit of using is that later when we want to change it to something else, we only need to change that using statement. If we have that using at the beginning, we don't need so many files update of the types.


I'm OK with this. I guess later if we want to update it globally to something else, maybe we can abstract it. That's beyond the scope of this one.

Reply:
>I agreed that instrumented_io_context does show more semantics about the purpose. The benefit of using is that later when we want to change it to something else, we only need to change that using statement. If we have that using at the beginning, we don't need so many files update of the types.

I totally get that reasoning, but since the users of `instrumented_io_context` are relying on an interface to use that instrumentation (`StatsString()` for an event loop stats summary, `post()` with an extra handler name argument), I think that having the abstraction's name convey the semantics of that interface is more of a good thing than a bad thing. If we change the semantics of the interface, then the name should be updated at that time to reflect those changed semantics, even if it has to be changed in a lot of places.

And thankfully code refactoring tools make these kind of name changes easy, even with basic shell tools:
```
find . -type f \( -name "*.h" -o -name "*.cc" \) -exec sed -i'' -e 's/instrumented_io_context/ray_io_context/g' {} +
```